data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateProcessor.java

```diff
@@ -125,6 +125,7 @@ public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
             }
             final IdentificationKeysHasher.IdentificationKeysMap identificationKeysMap = identificationKeysHasher.createIdentificationKeysMapFromEvent(event);
             final AggregateGroup aggregateGroupForEvent = aggregateGroupManager.getAggregateGroup(identificationKeysMap);
```

Comment:
Consider automating the attachment of an event to the AggregateGroup during object creation so that it's encapsulated within the AggregateGroupManager

```
public AggregateGroup getAggregateGroup(Map<Object, Object> identificationKeysMap, Event event) {
        AggregateGroup aggregateGroup = getOrCreateAggregateGroup(identificationKeysMap);
        ...
        aggregateGroup.attachToEventAcknowledgementSet(event);
        return aggregateGroup;
    }
```